api/apis/authentication_middleware.go

```diff
@@ -74,3 +76,14 @@ func (a *AuthenticationMiddleware) Middleware(next http.Handler) http.Handler {
 		next.ServeHTTP(w, r)
 	})
 }
+
+// Eventually we will want to authenticate the log-cache read endpoint and go back to using the simple
+// unauthenticatedEndpoints map. For now though we need to do a more complicated regexp match against
+// the path since the log-cache read endpoint includes an arbitrary guid as part of its path
+var logCacheReadEndpointRegexp = regexp.MustCompile(`\/api\/v1\/read\/[0-9a-fA-F\-]*$`)
```

Comment:
I'm open to suggestions here around other ways of doing this. The `unauthenticatedEndpoints` map didn't work in this case because `r.URL.Path` isn't a static path for this endpoint, but I wanted to preserve that map because I think we can go back to using it and it alone once we fix the CLI and can require authentication for this endpoint.

Reply:
I like the simplicity / performance of the map versus a composite regexp that could match all of these endpoints.

Reply:
I think this is fine as a temporary measure.